### PR TITLE
Retrieve Attestations from Operation Service

### PR DESCRIPTION
**Pull request by terenc3t**
This is part of #1565 

Key changes:
- Implemented functions to retrieve all the attestations and delete attestation in package db
- Implemented function to retrieve attestation up to max attestation amount and delete retrieved attestations in package operation as part of operation service receiver
- Implemented tests for the above

What's left:
- RPC server needs to interface with operation service to retrieve these attestations. `func (s *Service) Attestations() ([]*pb.Attestation, error)`

### CODE REVIEW

File: beacon-chain/db/attestation.go [hunk 1]
```diff
@@ -43,6 +54,28 @@ func (db *BeaconDB) Attestation(hash [32]byte) (*pb.Attestation, error) {
...
+	var attestations []*pb.Attestation
+	err := db.view(func(tx *bolt.Tx) error {
+		a := tx.Bucket(attestationBucket)
+
+		if err := a.ForEach(func(k, v []byte) error {
+			attestation, err := createAttestation(v)
+			if err != nil {
+				return err
+			}
+			attestations = append(attestations, attestation)
```
Comment (rauljordan) [inline beacon-chain/db/attestation.go]: Why don't we cap it off at max attestations here? What if we have a million records in the db, this method will fetch them all

Comment (terenc3t) [reply beacon-chain/db/attestation.go]: I'd rather keep this function as generic as possible so anyone can use it w/o having to worry it's capped or not. I think the clear separation of responsibility is nice (DB retrieves it, and Operation service caps it).
In terms of millions records in the DB, we have clean up routines that deletes attestation after it's seen from the block, and we only keep attestations up to last finalized block.

Comment (terenc3t) [reply beacon-chain/db/attestation.go]: Will update comment to be my explicit, these are the attestations that have not been included on chain

Comment (rauljordan) [reply beacon-chain/db/attestation.go]: Fair enough, I see the use case where we might want to fetch all of them perhaps for visualizations or metrics gathering

**Pull request merged**